tests/e2e/workloads/pvc_snapshot_and_clone/test_small_files_pvc_cloned_status.py

```diff
@@ -0,0 +1,153 @@
+import os
+import urllib.request
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.framework.testlib import bugzilla
+from ocs_ci.ocs.node import get_worker_nodes
+from ocs_ci.ocs.resources import pod, pvc
+from ocs_ci.helpers import helpers
+from tests.conftest import snapshot_factory
+from ocs_ci.framework.testlib import (
+    E2ETest,
+    tier2,
+)
+from ocs_ci.ocs.exceptions import CommandFailed
+
+log = logging.getLogger(__name__)
+
+
+@tier2
+class TestPvcClonedStatusOfSmallSizeFiles(E2ETest):
+    """
+    Tests to verify PVC status bound or not if
+    files size are smaller.
+    """
+
+    @bugzilla("2039265")
+    def test_pvc_status_after_clone(self, pvc_factory, teardown_factory):
+        """
+        1. Create 50Mib PVC
+        2. Attach PVC to an application pod
+        3. Download Linux kernel and untar it to the directory where PVC is mounted
+        4. Create a new PVC from the snapshot
+        5. Take a snapshot of the PVC.
+        6. Create a new PVC out of that snapshot.
+        7. Verify PVC state
+
+        Args:
+            pvc_factory: A fixture to create new pvc
+            teardown_factory: A fixture to destroy objects
+        """
+        pvc_size = 90
+        worker_nodes_list = get_worker_nodes()
+        kernel_url = "https://cdn.kernel.org/pub/linux/kernel/v4.x/linux-4.4.tar.xz"
+        download_path = "tmp"
+        dir_path = os.path.join(os.getcwd(), download_path)
+        file_path = os.path.join(dir_path, "linux-4.4.tar.xz")
+        if not os.path.exists(dir_path):
+            os.makedirs(dir_path)
+        urllib.request.urlretrieve(kernel_url, file_path)
+
+        # Create a RWX PVC
+        pvc_obj = pvc_factory(
+            interface=constants.CEPHFILESYSTEM,
+            access_mode=constants.ACCESS_MODE_RWX,
+            status=constants.STATUS_BOUND,
+            size=pvc_size,
+            size_unit="Mi",
+        )
+
+        # Confirm PVCs are Bound
+        log.info("Verifying the CephFS PVC are Bound")
+        helpers.wait_for_resource_state(
+            resource=pvc_obj, state=constants.STATUS_BOUND, timeout=240
+        )
+        pvc_obj.reload()
+        log.info("Verified: CephFS PVC are Bound")
+
+        # Create a pod on a particular node
+        selected_node = random.choice(worker_nodes_list)
+        log.info(f"Creating a pod on node: {selected_node} with pvc {pvc_obj.name}")
+
+        pod_obj = helpers.create_pod(
+            interface_type=constants.CEPHFILESYSTEM,
+            pvc_name=pvc_obj.name,
+            namespace=pvc_obj.namespace,
+            node_name=selected_node,
+            pod_dict_path=constants.PERF_POD_YAML,
+        )
+        teardown_factory(pod_obj)
+
+        # Confirm that the pod is running on the selected_node
+        helpers.wait_for_resource_state(
+            resource=pod_obj, state=constants.STATUS_RUNNING, timeout=180
+        )
+        pod_obj.reload()
+        assert pod.verify_node_name(
```

Comment:
why these steps are needed? As per the BZ, you dont need to push the pod to a particular node

Reply:
added this to verify on which node it is running to back trace if something wrong

Reply:
Okay. But only add it if it's necessary for the flow you're trying to achieve. otherwise its just extra lines of codes.